python/mxnet/contrib/onnx/mx2onnx/_op_translations.py

```diff
@@ -1841,20 +1840,24 @@ def convert_cast(node, **kwargs):
     """Map MXNet's Cast operator attributes to onnx's Cast operator
     and return the created node.
     """
-    helper, _, _ = import_onnx_modules()
+    onnx = import_onnx_modules()
     name = node["name"]
     proc_nodes = kwargs["proc_nodes"]
     inputs = node["inputs"]
     dtype = node["attrs"]["dtype"]
+    if dtype == 'float32':
+        dtype = 'float'
+    elif dtype == 'float64':
```

Comment:
whats the behavior for other types and should we handle default type?

Reply:
I think it is better if we do type = numpy.dtype(dtype) and later type.name(). What do you think?

Reply:
Please add comment on why float instead of float32 and double instead of float64.

Reply:
This is mapping from mxnet's dtype for cast operator to onnx's dtypes. Added comment to better explain.

TensorProto maps 'float32' to 'float' and 'float64' to 'double'. Rest of the types are handled by this line while creating cast op node->   " to=getattr(onnx.TensorProto, dtype.upper())",